polaris/evaluate/_predictions.py

```diff
@@ -177,3 +179,69 @@ def _is_fully_specified(
                 return False
 
         return True
+
+    def get_subset(
+        self, test_set_subset: list[str] | None = None, target_subset: list[str] | None = None
+    ) -> "BenchmarkPredictions":
+        """Return a subset of the original predictions"""
+
+        if test_set_subset is None and target_subset is None:
+            return self
+
+        if test_set_subset is None:
+            test_set_subset = self.test_set_labels
+        if target_subset is None:
+            target_subset = self.target_labels
+
+        predictions = defaultdict(dict)
+        for test_set_name in self.predictions.keys():
+            if test_set_subset is not None and test_set_name not in test_set_subset:
+                continue
+
+            for target_name, target in self.predictions[test_set_name].items():
+                if target_subset is not None and target_name not in target_subset:
```

Comment:
**Question**
Likewise here, there's a guard above (`if target_subset is None:  target_subset = self.target_labels`) that should ensure this is always `not None`.